js/private/js_image_layer.mjs

```diff
@@ -1,9877 +1,339 @@
-        configurable: true,
-        value: "AbortSignal",
-    });
-}
-
-/**
- * The AbortController.
- * @see https://dom.spec.whatwg.org/#abortcontroller
- */
-class AbortController {
-    /**
-     * Initialize this controller.
-     */
-    constructor() {
-        signals.set(this, createAbortSignal());
-    }
-    /**
-     * Returns the `AbortSignal` object associated with this object.
-     */
-    get signal() {
-        return getSignal(this);
-    }
-    /**
-     * Abort and signal to any observers that the associated activity is to be aborted.
-     */
-    abort() {
-        abortSignal(getSignal(this));
-    }
-}
-/**
- * Associated signals.
- */
-const signals = new WeakMap();
-/**
- * Get the associated signal of a given controller.
- */
-function getSignal(controller) {
-    const signal = signals.get(controller);
-    if (signal == null) {
-        throw new TypeError(`Expected 'this' to be an 'AbortController' object, but got ${controller === null ? "null" : typeof controller}`);
+        })
+        resolveTasks.push(resolveTask)
     }
-    return signal;
-}
-// Properties should be enumerable.
-Object.defineProperties(AbortController.prototype, {
-    signal: { enumerable: true },
-    abort: { enumerable: true },
-});
-if (typeof Symbol === "function" && typeof Symbol.toStringTag === "symbol") {
-    Object.defineProperty(AbortController.prototype, Symbol.toStringTag, {
-        configurable: true,
-        value: "AbortController",
-    });
-}
-
-var abortController = /*#__PURE__*/Object.freeze({
-	__proto__: null,
-	'default': AbortController,
-	AbortController: AbortController,
-	AbortSignal: AbortSignal
-});
-
-var require$$9 = /*@__PURE__*/getAugmentedNamespace(abortController);
-
-var errors;
-var hasRequiredErrors;
-
-function requireErrors () {
-	if (hasRequiredErrors) return errors;
-	hasRequiredErrors = 1;
-
-	const { format, inspect, AggregateError: CustomAggregateError } = requireUtil();
-
-	/*
-	  This file is a reduced and adapted version of the main lib/internal/errors.js file defined at
-
-	  https://github.com/nodejs/node/blob/master/lib/internal/errors.js
 
-	  Don't try to replace with the original file and keep it up to date (starting from E(...) definitions)
-	  with the upstream file.
-	*/
+    await Promise.all(resolveTasks);
 
-	const AggregateError = globalThis.AggregateError || CustomAggregateError;
-	const kIsNodeError = Symbol('kIsNodeError');
-	const kTypes = [
-	  'string',
-	  'function',
-	  'number',
-	  'object',
-	  // Accept 'Function' and 'Object' as alternative to the lower cased version.
-	  'Function',
-	  'Object',
-	  'boolean',
-	  'bigint',
-	  'symbol'
-	];
-	const classRegExp = /^([A-Z][a-z0-9]*)+$/;
-	const nodeInternalPrefix = '__node_internal_';
-	const codes = {};
-	function assert(value, message) {
-	  if (!value) {
-	    throw new codes.ERR_INTERNAL_ASSERTION(message)
-	  }
-	}
+    await Promise.all([
```

Comment:
If we're doing `await` on the `WRITE_STATEMENTS` shouldn't we also do `await` on the `{}unusedinputs.close()` calls?

Otherwise we may as well just remove this `Promise.all` since we're not waiting for "all" things anyway.

Reply:
we can't since we need stat calls to finish before we can sort the array. so this is working as intended.

Reply:
Should we also `await` all the `*unusedinputs.close()` or do you want to just assume they are written before node exits? I think we can assume it but I thought it was clearer when we close them, see [CLOSE_STATEMENTS](https://github.com/jbedard/rules_js/blob/daf19490b832c87486dde357be3c6d5cc9a78f7d/js/private/js_image_layer.bzl#L356C9-L356C25) in my branch.

Reply:
they are closed automatically.